bindings/python/triplestore/backends/ontopy.py

```diff
@@ -0,0 +1,197 @@
+
+        The source is specified using one of `source`, `location` or `data`.
+
+        Parameters:
+            source: File-like object or file name.
+            location: String with relative or absolute URL to source.
+            data: String containing the data to be parsed.
+            format: Needed if format can not be inferred from source.
+            encoding: Encoding argument to io.open().
+            kwargs: Additional keyword arguments passed to Ontology.load().
+        """
+        if source:
+            self.onto.load(filename=source, format=format, **kwargs)
+        elif location:
+            self.onto.load(filename=location, format=format, **kwargs)
+        elif data:
+            #s = io.StringIO(data)
+            #self.onto.load(filename=s, format=format, **kwargs)
+
+            # Could have been done much nicer if it hasn't been for Windows
+            filename = None
+            try:
+                kw = {"delete": False}
+                if isinstance(data, str):
+                    kw.update(mode="w+t", encoding=encoding)
+                with tempfile.NamedTemporaryFile(**kw) as f:
+                    f.write(data)
+                    filename = f.name
+                self.onto.load(filename=filename, format=format, **kwargs)
+            finally:
+                if filename:
+                    os.remove(filename)
+
+        else:
+            raise ValueError(
+                "either `source`, `location` or `data` must be given"
+            )
+
+    def serialize(self, destination=None, format='turtle', **kwargs):
+        """Serialise to destination.
+
+        Parameters:
+            destination: File name or object to write to.  If None, the
+                serialisation is returned.
+            format: Format to serialise as.  Supported formats, depends on
+                the backend.
+            kwargs: Passed to the Ontology.save() method.
+
+        Returns:
+            Serialised string if `destination` is None.
+        """
+        if destination:
+            self.onto.save(destination, format=format, **kwargs)
+        else:
+            # Clumsy implementation due to Windows...
+            filename = None
+            try:
+                with tempfile.NamedTemporaryFile(delete=False) as f:
+                    filename = f.name
+                    self.onto.save(filename, format=format, **kwargs)
+                with open(filename, 'rt') as f:
+                    return f.read()
+            finally:
+                if filename:
+                    os.remove(filename)
+
+    def query(self, query_object, native=True, **kwargs):
+        """SPARQL query."""
+        if native:
+            res = self.onto.world.sparql(query_object)
+        else:
+            graph = self.onto.world.as_rdflib_graph()
+            res = graph.query(query_object, **kwargs)
+        # TODO: Convert result to expected type
```

Comment:
Is this TODO important?

Reply:
The TODO should be removed and placed into a github issue

Reply:
Agree that it should be made an issue, but I don't think that we should remove the TODO - it makes it easier to spot where the needed changes should go

Reply:
Added issue #336